### PR TITLE
Added a drupal alter in xml_form_builder_get_form to allow for the form ...

### PR DESCRIPTION
**Pull request by matthewperry**
Added a drupal alter in xml_form_builder_get_form to allow for the form builder schema to be modified after it's been added to the form array.  This is needed for the metrodora module to allow it to modify tab panel access permission to the form builder schema that's saved into the form array.

### CODE REVIEW

**Comment by adam-vessey** [comment]
This shouldn't really be necessary. Should be able to do a general `hook_form_alter()`, and figure out whether or not we need to make adjustments...

File: builder/xml_form_builder.module [hunk 1]
```diff
@@ -745,25 +757,3 @@ function xml_form_builder_islandora_derivative(AbstractObject $object = NULL) {
...
-    'default_hook' => 'xml_form_builder_default_dc_config',
-    'feature_source' => TRUE,
-  );
-  $items['xml_form_builder_default_dc_xslt'] = array(
-    'name' => t('Islandora XML Form Builder Default DC Transformation XSLTs'),
-    'feature_source' => FALSE,
-  );
-
-  return $items;
-}
```
Comment (adam-vessey) [inline builder/xml_form_builder.module:769]: Bad resolution of merge conflict? Should probably avoid nuking...

**Comment by nigelgbanks** [comment]
> This shouldn't really be necessary. Should be able to do a general hook_form_alter(), and figure out  whether or not we need to make adjustments...

It is actually necessary with the form builder definition, alters get run after the definition of the form is put into the form_state. So they don't "Just Work" in that you'd also have to change the form_state as well, also the application of the XML document to the form (creates new elements dynamically) happens before the alter. This is a much easier solution to that problem.

File: builder/xml_form_builder.module (hunk 1 above)
Comment (matthewperry) [inline builder/xml_form_builder.module:769]: Thanks, put back nuked function.

**Comment by adam-vessey** [comment]
> > This shouldn't really be necessary. Should be able to do a general hook_form_alter(), and figure out  whether or not we need to make adjustments...
> 
> It is actually necessary with the form builder definition, alters get run after the definition of the form is put into the form_state. So they don't "Just Work" in that you'd also have to change the form_state as well, also the application of the XML document to the form (creates new elements dynamically) happens before the alter. This is a much easier solution to that problem.

I was talking about the original problem being addressed in particular, which was (and still is) partially being handled with a `hook_form_alter()`...

The original stuff I'm talking about is:
- [`metrodora_form_alter()`](https://github.com/discoverygarden/metrodora/blob/dcc65cecd538272ea3565a287c0e22ca9a172885/metrodora.module#L253-L278)
- [`metrodora_modify_namespace_specific_fields()`](https://github.com/discoverygarden/metrodora/blob/dcc65cecd538272ea3565a287c0e22ca9a172885/metrodora.module#L346-L367)
- [`metrodora_build_variable_elements_array()`](https://github.com/discoverygarden/metrodora/blob/dcc65cecd538272ea3565a287c0e22ca9a172885/metrodora.module#L416-L462)
- [`metrodora_build_repeatable_element_array()`](https://github.com/discoverygarden/metrodora/blob/dcc65cecd538272ea3565a287c0e22ca9a172885/metrodora.module#L322-L334)

Not certain, but pretty sure a slight adjustment in how the tabs/tabpanel elements were being processed could have addressed the issue.

**Pull request merged**